snuba/datasets/entities/discover.py

```diff
@@ -492,7 +492,7 @@ def get_query_processors(self) -> Sequence[QueryProcessor]:
             apdex_processor(columnset),
             failure_rate_processor(columnset),
             HandledFunctionsProcessor("exception_stacks.mechanism_handled", columnset),
-            TimeSeriesColumnProcessor({"time": "timestamp"}),
+            TimeSeriesProcessor(self.__time_group_columns, self.__time_parse_columns),
```

Comment:
I think we should move the condition processing early on in the list of query processors. That processor essentially properly parses a datetime literal into a datetime object. All query processors should have access to the result.
I'd say you can just move the whole processor to the beginning of the list.